src/pubsub/ua_pubsub_ns0.c

```diff
@@ -1792,6 +1898,31 @@ addDataSetWriterRepresentation(UA_Server *server, UA_DataSetWriter *dataSetWrite
                             UA_NODEID_NUMERIC(0, UA_NS0ID_HASPROPERTY),
                             dataSetWriter->identifier);
 
+    UA_NodeId statusIdNode = 
+        findSingleChildNode(server, UA_QUALIFIEDNAME(0, "Status"),
+                            UA_NODEID_NUMERIC(0, UA_NS0ID_HASCOMPONENT),
+                            dataSetWriter->identifier);
+    
+    if(UA_NodeId_isNull(&statusIdNode)) {
+        return UA_STATUSCODE_BADNOTFOUND;
+    }
+
+    UA_NodeId stateIdNode = 
+        findSingleChildNode(server, UA_QUALIFIEDNAME(0, "State"),
+                            UA_NODEID_NUMERIC(0, UA_NS0ID_HASCOMPONENT),
+                            statusIdNode);
+
+    if(UA_NodeId_isNull(&keyFrameNode)) {
+
+        // TODO: The keyFrameNode is NULL here, does not depend on the pubsub changes
```

Comment:
If we do nothing here, a simple comment is fine.

Reply:
Okay